docs/BypassingModuleMocks.md

```diff
@@ -0,0 +1,60 @@
+---
+id: bypassing-module-mocks
+title: Bypassing module mocks
+---
+
+Jest allows you to mock out whole modules in your tests, which can be useful for testing your code is calling functions from that module correctly. However, sometimes you may want to use parts of a mocked module in your _test file_, in which case you want to access the original implementation, rather than a mocked version.
+
+Consider writing a test case for this `createUser` function:
+
+```javascript
+// createUser.js
+import fetch from 'node-fetch';
+
+export const createUser = async () => {
+  const response = await fetch('http://website.com/users', {method: 'POST'});
+  const userId = await response.text();
+  return userId;
+};
+```
+
+Your test will want to mock the `fetch` function so that we can be sure that it gets called without actually making the network request. However, you'll also need to mock the return value of `fetch` with a `Response` (wrapped in a `Promise`), as our function uses it to grab the created user's ID. So you might initially try writing a test like this:
+
+```javascript
+jest.mock('node-fetch');
+
+import fetch, {Response} from 'node-fetch';
+
+import {createUser} from './createUser';
+
+test('createUser calls fetch with the right args and returns the user id', async () => {
+  fetch.mockReturnValue(Promise.resolve(new Response('4')));
```

Comment:
Could use `fetch.mockResolvedValue(new Response('4'))` here

Reply:
Nice, didn't know that existed!